sockeye/decoder.py

```diff
@@ -251,7 +251,7 @@ def forward(self, step_input: pt.Tensor, states: List[pt.Tensor]) -> Tuple[pt.Te
             autoregr_states = [list(islice(states_iter, 0, layer.num_state_tensors)) for layer in self.layers]  # type: ignore
 
         batch, heads, target_max_len, source_max_len = source_mask.size()
-        source_mask_view = source_mask.view(batch * heads, target_max_len, source_max_len)
+        source_mask_view = source_mask.reshape(batch * heads, target_max_len, source_max_len)
```

Comment:
Have you measured the effect of using `reshape()` instead of `view()` on CPU/GPU latency?

If we make this change, it would also make sense to rename the variable to `source_mask_reshape`.

Reply:
Thanks Michael! I just changed the variable to `source_mask_reshape` as suggested.

Regarding speed, using reshape instead of view increases the latency to `4%` on GPU (K80) and `11%` on CPU.
Note that this reshaping is called only `1` time for each batch. Meanwhile repeat_interleave_with_expand is called a lot more than that (I noticed it is around `50` times for a batch I tried).
Code comparison:

```
import timeit
set_up_line = """
import torch as pt
pt.manual_seed(1)
heads = 8
batch = 20
target_max_len = source_max_len = 50
a = pt.rand(20, 8, 50, 50)
"""
executive_line = """
b = a.reshape(batch * heads, target_max_len, source_max_len)
"""
timeit.timeit(executive_line,setup=set_up_line, number=10000)
#cpu: 0.031618487962987274
#gpu: 0.013913217990193516

executive_line = """
b = a.view(batch * heads, target_max_len, source_max_len)
"""
timeit.timeit(executive_line,setup=set_up_line, number=10000)
#cpu: 0.028475572005845606
#gpu: 0.013381681987084448
```